app/javascript/shared/components/focustrap.jsx

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import { h, Fragment } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import { createFocusTrap } from 'focus-trap';
+import { defaultChildrenPropTypes } from '../../common-prop-types';
+
+const FocusTrap = ({ selector, children }) => {
+  const focusTrap = useRef(null);
+
+  useEffect(() => {
+    focusTrap.current = createFocusTrap(selector);
+    focusTrap.current.activate();
+    return () => {
+      focusTrap.current.deactivate();
+    };
+  });
+
+  return <Fragment>{children}</Fragment>;
+};
+
+FocusTrap.defaultProps = {
+  selector: '.crayons-modal',
```

Comment:
I made the default selector value `'.crayon-modal'` since this component will be mostly be used by modals (rarely in other places where different selector string could be passed)